File: 프로그래머스/2/42883. 큰 수 만들기/큰 수 만들기.py
def solution(number, k):
    answer = []
    for num in number:
        while(k>0 and answer and answer[-1] < num):
            answer.pop()
            k-=1
        answer.append(num)
    while(k>0):
        answer.pop()
        k-=1
    
    return ''.join(answer)

File: 프로그래머스/2/42883. 큰 수 만들기/큰 수 만들기.js
function solution(number, k) {
    var answer = '';
    num_lst=Array.from(number).map(Number)
    answer = []

    for(let num of num_lst){
        if (answer.length === 0){
            answer.push(num)
            continue
        }
        while (k>0 && answer[answer.length-1] < num){
            answer.pop()
            k--
        } 
        answer.push(num)
        
    }
    
    while(k>0){
        answer.pop()
        k--
    }
    
    return answer.join('');
}

console.log(solution("22",1))